### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by maksym-bakunov**

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/OnlineBookStoreApplication.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.example.onlinebookstore;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class OnlineBookStoreApplication {
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/OnlineBookStoreApplication.java]:
```suggestion
```

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.util.Currency;
+import lombok.Data;
+import lombok.NonNull;
+
+@Entity
+@Data
+@Table(name = "book")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/Book.java]: Tables should be names as plural

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/Book.java]: it's redundant

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    private String isbn;
+    @Column(name = "price")
+    @NonNull
+    private Currency price;
+    @Column(name = "description")
+    private String description;
+    @Column(name = "cover_image")
+    private String coverImage;
+
+    public Book() {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/Book.java]: Why do you need this constructor?

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    @Column(name = "title")
+    private String title;
+    @NonNull
+    @Column(name = "author")
+    private String author;
+    @Column(name = "isbn", unique = true)
+    private String isbn;
+    @Column(name = "price")
+    @NonNull
+    private Currency price;
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/Book.java]:
```suggestion
    private BigDecimal price;
```

File: src/main/java/com/example/onlinebookstore/service/impl/BookServiceImp.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.service.BookService;
+import java.util.List;
+
+public class BookServiceImp implements BookService {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/service/impl/BookServiceImp.java]:
```suggestion
public class BookServiceImpl implements BookService {
```

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,7 @@
+spring.datasource.url=jdbc:mysql://localhost:3306/book_store
+spring.datasource.username=root
+spring.datasource.password=ps1234
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+
+spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.show-sql = true
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties:7]:
```suggestion
spring.jpa.show-sql=true
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
It looks like your project tree is wrong. pom file and gitignore file should be below of src package.

**Comment by maksym-bakunov** [comment]
> It looks like your project tree is wrong. pom file and gitignore file should be below of src package.

What do you mean? Bellow is the screen of my project:
https://c2n.me/4kI0ICr

File: src/main/java/com/example/onlinebookstore/OnlineBookStoreApplication.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class OnlineBookStoreApplication {
+    public static void main(String[] args) {
+        SpringApplication.run(OnlineBookStoreApplication.class, args);
+    }
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/onlinebookstore/OnlineBookStoreApplication.java:11]:
```suggestion
```

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**